aips/account_abstraction.md

```diff
@@ -0,0 +1,213 @@
+    account: AccountAddress,
+    function_info: FunctionInfo,
+    auth_data: &AbstractionAuthData,
+    traversal_context: &mut TraversalContext,
+    module_storage: &impl AptosModuleStorage,
+) -> VMResult<Vec<u8>> {
+    let auth_data = bcs::to_bytes(auth_data).expect("from rust succeeds");
+    let mut params =
+        serialize_values(&vec![
+            MoveValue::Signer(account),
+            function_info.as_move_value(),
+        ]);
+    params.push(auth_data);
+    session
+        .execute_function_bypass_visibility(
+            &LITE_ACCOUNT_MODULE,
+            AUTHENTICATE,
+            vec![],
+            params,
+            gas_meter,
+            traversal_context,
+            module_storage,
+        )
+        .map(|mut return_vals| {
+            assert!(
+                return_vals.mutable_reference_outputs.is_empty()
+                    && return_vals.return_values.len() == 1,
+                "Abstraction authentication function must only have 1 return value"
+            );
+            let (signer_data, signer_layout) = return_vals.return_values.pop().expect("Must exist");
+            assert_eq!(
+                signer_layout,
+                MoveTypeLayout::Signer,
+                "Abstraction authentication function returned non-signer."
+            );
+            signer_data
+        })
+}
+```
+
+### Authentication Flow
+The AA authentication flow works as below:
+1. Transaction carries the Abstraction Authenticator.
+2. Rust authenticator will run and find it is Abstraction Authenticator so native authentication is noop. But `authentication_key` will be set to `None`.
+3. When running prologue, if the `authentication_key` is `None`, it will bypass the authentication key check.
+4. After the prologue, AptosVM will check each senders' whether their account authenticator is Abstraction Authenticator. If yes, AptosVM will run the move auth function specified in the `function_info` with `auth_data` and expect a `signer` to be returned. If any sender's auth function aborts, the authentication check for the whole transaction fails. It is noted that this function has a gas limit if the function call runs above the gas limit it will abort immediately.
+5. Run the user transaction body as usual with the `signer`s just returned.
+
+## Reference Implementation
+https://github.com/aptos-labs/aptos-core/pull/15219
+
+The feature flag for this feature is `std::features::ACCOUNT_ABSTRACTION`.
+
+## Testing
+The above PR has API e2e test the functionality of Account abstraction with various cases.
+
+## Risks and Drawbacks
+
+- AA is hard to adopt w/o an auth function library. To make it easy for user to use, the framework should provide a bunch of commonly used 
+authentication modules/functions for users to use. Otherwise, it is really hard for beginner to implement customized
+authentication logic in move.
+
+- The gas limit of the auth function is fixed so AA cannot support complicated move functions.
```

Comment:
maybe `cannot support complicated` => `cannot support too complicated`, as everything is relative :)